bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/target/env/AsterismType.java

```diff
@@ -2,38 +2,58 @@
 
 import edu.gemini.pot.sp.SPComponentBroadType;
 import edu.gemini.pot.sp.SPComponentType;
-import edu.gemini.shared.util.immutable.DefaultImList;
-import edu.gemini.shared.util.immutable.ImList;
+import edu.gemini.shared.util.immutable.*;
+import edu.gemini.spModel.gemini.ghost.AsterismConverters;
 import edu.gemini.spModel.gemini.ghost.GhostAsterism$;
 
+import java.util.function.Supplier;
+
 public enum AsterismType {
-    Single("single") {
-        @Override
-        public Asterism createEmptyAsterism() {
-            return Asterism$.MODULE$.createSingleAsterism();
-        }
-    },
-    GhostStandardResolution("ghostStandardRes") {
-        @Override
-        public Asterism createEmptyAsterism() {
-            return GhostAsterism$.MODULE$.createEmptyStandardResolutionAsterism();
-        }
-    },
-    GhostHighResolution("ghostHighRes") {
-        @Override
-        public Asterism createEmptyAsterism() {
-            return GhostAsterism$.MODULE$.createEmptyHighResolutionAsterism();
-        }
-    },
+    /**
+     * Note all GHOST standard resolution asterisms have the same tag because we use this to output them to XML
+     * in a category together.
+     */
+    Single("single", "Single Target", None.instance(), Asterism$.MODULE$::createSingleAsterism),
+
+    GhostStandardResolutionSingleTarget("ghostStandardRes", "Single Target",
+            new Some<>(AsterismConverters.GhostSingleTargetConverter$.MODULE$),
+            GhostAsterism$.MODULE$::createEmptyStandardResolutionSingleTargetAsterism),
+
+    GhostStandardResolutionDualTarget("ghostStandardRes", "Dual Target",
+            new Some<>(AsterismConverters.GhostDualTargetConverter$.MODULE$),
+            GhostAsterism$.MODULE$::createEmptyStandardResolutionDualTargetAsterism),
+
+    GhostStandardResolutionTargetPlusSky("ghostStandardRes", "SRIFU1 Target, SRIFU2 Sky",
+            new Some<>(AsterismConverters.GhostTargetPlusSkyConverter$.MODULE$),
+            GhostAsterism$.MODULE$::createEmptyStandardResolutionTargetPlusSkyAsterism),
+
+    GhostStandardResolutionSkyPlusTarget("ghostStandardRes", "SRIFU1 Sky, SRIFU2 Target",
+            new Some<>(AsterismConverters.GhostSkyPlusTargetConverter$.MODULE$),
+            GhostAsterism$.MODULE$::createEmptyStandardResolutionSkyPlusTargetAsterism),
+
+    GhostHighResolution("ghostHighRes", "High Resolution",
+            new Some<>(AsterismConverters.GhostHighResolutionConverter$.MODULE$),
+            GhostAsterism$.MODULE$::createEmptyHighResolutionAsterism),
     ;
 
     public final String tag;
+    public final Option<AsterismConverters.AsterismConverter> converter;
```

Comment:
Something about the `converter` is troubling me.  In general there are _O(n!)_ combinations of pairs of asterisms that you could want to convert.  I suppose most of the conversions are just not specified though and return `None`?

Reply:
How do you see O(n!) ?

I would think it would be O(n^2), as we'd want to convert between any pair of asterism types, so assuming n asterism types, nC2 = n(n-1)/2 ?

Reply:
Actually, nix that. It should be O(n^2), but n(n-1) and not divided by two, obviously.